sympy/assumptions/ask.py

```diff
@@ -497,6 +498,10 @@ def ask(proposition, assumptions=True, context=global_assumptions):
 
     # using satask (still costly)
     res = satask(proposition, assumptions=assumptions, context=context)
+    if res is not None:
+        return res
+
+    res = lra_satask(proposition, assumptions=assumptions, context=context)
```

Comment:
I would suggest refactoring this so that lra_satask raises UnhandledInput exceptions, and the logic here catches those exceptions and returns None. That way, you can call lra_satask directly if you want to be clear about whether something is returning None because it is actually undetermined or just because the input currently can't be handled.